### ISSUE TITLE
SNOW-809206: Add setup instructions for contributors

### ISSUE THREAD

**Issue by saulbein** [snowflakedb/snowflake-sqlalchemy#412]
## What is the current behavior?
As it stands, this repository is quite difficult to contribute to as there are no instructions on how to set up a local environment to run all tests, as the tox configuration is meant for some managed? platform and doesn't work for local development. The tests folder has extremely basic instructions on how to prepare a `parameters.py` file for authenticate with Snowflake, but even that is flawed as it doesn't support other authentication methods other than password-based (I use my company's custom authenticator and had to change a lot of code to make the tests succeed).

## What is the desired behavior?
Documentation included that would show how to set up this project for local development.

## How would this improve `snowflake-sqlalchemy`?
It would allow developers who have the skills to contribute fixes to the project much easier, so they can get the features they want faster by implementing them themselves.

## References, Other Background
I have contributed multiple PRs to this project already and so far setting up the environment has always been the most complicated part of it.

**Comment by sfc-gh-sfan** [comment]
@saulbein : We are looking at it and our current ETA is end of July. Thanks for the suggestion :)

**Comment by fredrike** [comment]
> @saulbein : We are looking at it and our current ETA is end of July. Thanks for the suggestion :)

What year?

**Comment by sfc-gh-dszmolka** [comment]
indeed, estimations from 2023 are probably not valid in 2024 anymore, so currently there is no timeline attached to addressing this request. All the resources are dedicated to more burning matters, like implementing SQLAlchemy 2.0 support which is perhaps we can agree, more important.

That said, if you have the possibility to submit a PR , you could help greatly in moving this request forward !

**Comment by sfc-gh-mraba** [comment]
Contributor setup and local development instructions now live in [CONTRIBUTING.md](https://github.com/snowflakedb/snowflake-sqlalchemy/blob/main/CONTRIBUTING.md) (environment setup, Hatch usage, and how to run the tests).

Closing as resolved — please open a follow-up if anything in the setup steps is missing or unclear.
